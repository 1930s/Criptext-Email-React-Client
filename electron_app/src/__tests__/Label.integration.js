/* eslint-env node, jest */

const DBManager = require('../DBManager');
const fs = require('fs');

beforeAll(async () => {
  await DBManager.createTables();
});

describe('Test TABLE[Label]::', () => {
  it('should create label to db', async () => {
    await DBManager.createLabel({
      color: '#ffffff',
      text: 'Inbox'
    });

    const labels = await DBManager.getAllLabels();
    expect(labels).toMatchSnapshot();
  });

  it('should update label: color and text', async () => {
    await DBManager.updateLabel({
      id: 1,
      color: '#000000',
      text: 'labelmodified'
    });
    const labels = await DBManager.getAllLabels();
    expect(labels).toMatchSnapshot();
  });

  it('should update label: color', async () => {
    await DBManager.updateLabel({ id: 1, color: '#111111' });
    const labels = await DBManager.getAllLabels();
    expect(labels).toMatchSnapshot();
  });

  it('should update label: text', async () => {
    await DBManager.updateLabel({ id: 1, text: 'labelmodified2' });
    const labels = await DBManager.getAllLabels();
    expect(labels).toMatchSnapshot();
  });

  afterAll(() => {
    fs.unlinkSync('./src/__tests__/test.db');
  });
});
