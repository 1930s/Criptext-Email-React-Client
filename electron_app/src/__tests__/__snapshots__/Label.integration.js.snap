// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test TABLE[Label]:: should create label to db 1`] = `
Array [
  Object {
    "color": "#ffffff",
    "id": 1,
    "text": "Inbox",
  },
]
`;

exports[`Test TABLE[Label]:: should update label: color 1`] = `
Array [
  Object {
    "color": "#111111",
    "id": 1,
    "text": "labelmodified",
  },
]
`;

exports[`Test TABLE[Label]:: should update label: color and text 1`] = `
Array [
  Object {
    "color": "#000000",
    "id": 1,
    "text": "labelmodified",
  },
]
`;

exports[`Test TABLE[Label]:: should update label: text 1`] = `
Array [
  Object {
    "color": "#111111",
    "id": 1,
    "text": "labelmodified2",
  },
]
`;
